src/main/java/uk/gov/companieshouse/lookup/controller/CompanyLookupController.java

```diff
@@ -1,50 +1,81 @@
 import uk.gov.companieshouse.lookup.service.CompanyLookupService;
+import uk.gov.companieshouse.lookup.validation.ValidationError;
+import uk.gov.companieshouse.lookup.validation.ValidationHandler;
 
 @Controller
-@RequestMapping("/accounts/lookup")
+@RequestMapping("/company-lookup/search")
 public class CompanyLookupController {
 
-    private static final UriTemplate FOUND_REDIRECT = new UriTemplate("/transaction/{transactionId}/confirmation");
+    private static final UriTemplate FOUND_REDIRECT = new UriTemplate(
+        "/company-lookup/{companyNumber}/confirmation");
+    private static final String COMPANY_LOOKUP = "lookup/companyLookup";
 
     @Autowired
     private CompanyLookupService companyLookupService;
 
-    @GetMapping
-    public String getCompanyLookup(Model model, HttpServletRequest request) throws ServiceException {
+    @Autowired
+    private ValidationHandler validationHandler;
 
-        model.addAttribute("companyLookup", new CompanyLookup());
-        model.addAttribute("backButton", "/nowhere");
-        return "lookup/companyLookup";
+    @GetMapping
+    public String getCompanyLookup(@RequestParam("forward") String forward, Model model) {
+        CompanyLookup companyLookup = new CompanyLookup();
+        model.addAttribute("companyLookup", companyLookup);
+        return COMPANY_LOOKUP;
     }
 
     @PostMapping
-    public String postCompanyLookup(@ModelAttribute("companyLookup") @Valid CompanyLookup companyLookup,
-        BindingResult bindingResult, Model model, HttpServletRequest request) throws ServiceException {
+    public String postCompanyLookup(@RequestParam("forward") String forward,
+        @ModelAttribute("companyLookup") @Valid CompanyLookup companyLookup,
+        BindingResult bindingResult, RedirectAttributes attributes) throws URIValidationException,
+        ApiErrorResponseException {
+
+        List<ValidationError> validationErrors = companyLookupService
+            .validateCompanyLookup(companyLookup);
+        if (!validationErrors.isEmpty()) {
+            validationHandler.bindValidationErrors(bindingResult, validationErrors);
+            return COMPANY_LOOKUP;
+        }
 
-        CompanyProfileApi companyProfile = companyLookupService
-            .getCompanyProfile(companyLookup.getCompanyId());
+        try {
+            CompanyConfirmation companyConfirmation = companyLookupService
+                .getCompanyProfile(companyLookup.getCompanyNumber());
+            attributes.addAttribute("forward", forward);
+            attributes.addFlashAttribute("companyConfirmation", companyConfirmation);
 
-        model.addAttribute("companyProfile", companyProfile);
+            return UrlBasedViewResolver.REDIRECT_URL_PREFIX +
+                FOUND_REDIRECT.expand(companyLookup.getCompanyNumber()).toString();
 
-        return UrlBasedViewResolver.REDIRECT_URL_PREFIX +
-            FOUND_REDIRECT.expand("").toString();
+        } catch (ApiErrorResponseException e) {
```

Comment:
Could this logic be handled within the service layer? Ideally the controller wouldn't care about constructing validation errors - that should be the job of the validator class.